code/modules/mob/living/carbon/xenomorph/castes/widow/abilities_widow.dm

```diff
@@ -0,0 +1,295 @@
+
+	for(var/spawned_spiderlings = 0, spawned_spiderlings < amount_of_spiderlings, spawned_spiderlings++)
+		var/mob/living/carbon/xenomorph/spiderling/new_spiderling = new /mob/living/carbon/xenomorph/spiderling(current_controlling_spiderling.loc, current_controlling_spiderling)
+		/// here we add the created spiderligns to the list
+		create_spiderling_action.add_spiderling(new_spiderling)
+
+	create_spiderling_action.add_spiderling(current_controlling_spiderling)
+	succeed_activate()
+	add_cooldown()
+
+/// This happens whenever the spiderling that we control dies and there are spiderlings left to control
+/datum/action/xeno_action/spider_swarm/proc/switch_to_next_spiderling(mob/living/carbon/xenomorph/spiderling/spiderling)
+	SEND_SIGNAL(current_controlling_spiderling, COMSIG_ESCORTED_ATOM_CHANGING, spiderling)
+	current_controlling_spiderling.mind.transfer_to(spiderling)
+	current_controlling_spiderling = spiderling
+
+/// Put the player back in widow
+/datum/action/xeno_action/spider_swarm/proc/switch_to_mother()
+	owner.forceMove(get_turf(current_controlling_spiderling))
+	current_controlling_spiderling.mind.transfer_to(owner)
+
+// ***************************************
+// *********** Burrow
+// ***************************************
+
+/datum/action/xeno_action/burrow
+	name = "Burrow"
+	ability_name = "Burrow"
+	mechanics_text = " Burrow into the ground to hide in plain sight "
+	action_icon_state = "burrow"
+	plasma_cost = 1
+	cooldown_timer = 1 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_BURROW
+/datum/action/xeno_action/burrow/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(X.widow_burrowed)
+		return
+
+	/// We need the list of spiderlings so that we can burrow them
+	var/datum/action/xeno_action/create_spiderling/create_spiderling_action = owner.actions_by_path[/datum/action/xeno_action/create_spiderling]
+	to_chat(X, span_xenowarning("We start burrowing into the ground"))
+	if(!do_after(X, 3 SECONDS, TRUE, target, BUSY_ICON_DANGER))
+		return fail_activate()
+	to_chat(X, span_xenowarning("We have burrowed ourselves, we are hidden from the enemy"))
+	/// This is the burrow code for Widow
+	X.alpha = 0
+	X.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	X.density = FALSE
+	X.throwpass = TRUE
+	X.widow_burrowed = TRUE
+	/// Here we make every single spiderling that we have also burrow and assign a signal so that they unburrow too
+	for(var/mob/living/carbon/xenomorph/spiderling/spiderling in create_spiderling_action.spiderlings)
+		/// Here we register them for the signal and burrow them
+		RegisterSignal(spiderling, COMSIG_MOVABLE_MOVED, .proc/unburrow)
+		spiderling.alpha = 0
+		spiderling.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+		spiderling.density = FALSE
+		spiderling.throwpass = TRUE
+		spiderling.spiderling_burrowed = TRUE
+	/// This signal ensures that widow will also unborrow
+	RegisterSignal(X, COMSIG_MOVABLE_MOVED, .proc/unburrow)
```

Comment:
that means we move before the unburrow ? 

I doubt this will be bug less since it means we move with density and throwpass changed